identity/roles_test.go

```diff
@@ -11,38 +11,291 @@ package identity_test
 import (
 	"context"
 	"errors"
+	"fmt"
 	"io/ioutil"
+	"log"
 	"net/http"
 	"path"
 	"strings"
 	"testing"
+	"time"
 
+	triton "github.com/joyent/triton-go"
 	"github.com/joyent/triton-go/identity"
 	"github.com/joyent/triton-go/testutils"
 )
 
+const fakeRoleID = "e53b8fec-e661-4ded-a21e-959c9ba08cb2"
+
 var (
-	fakeRoleId          = "1234562335"
-	listRolesErrorType  = errors.New("unable to list roles")
-	getRoleErrorType    = errors.New("unable to get role")
-	deleteRoleErrorType = errors.New("unable to delete role")
-	createRoleErrorType = errors.New("unable to create role")
-	updateRoleErrorType = errors.New("unable to update role")
+	listRolesErrorType   = errors.New("unable to list roles")
+	getRoleErrorType     = errors.New("unable to get role")
+	deleteRoleErrorType  = errors.New("unable to delete role")
+	createRoleErrorType  = errors.New("unable to create role")
+	updateRoleErrorType  = errors.New("unable to update role")
+	setRoleTagsErrorType = errors.New("unable to set role tags")
+	getRoleTagsErrorType = errors.New("unable to get role tags")
 )
 
+func TestAccIdentity_SetRoleTags(t *testing.T) {
+	testRoleName := testutils.RandPrefixString("TestAccSetRoleTags", 32)
+
+	// Holds newly created role.
+	var newRole *identity.Role
+
+	testutils.AccTest(t, testutils.TestCase{
+		Steps: []testutils.Step{
+			&testutils.StepClient{
+				StateBagKey: "identity",
+				CallFunc: func(config *triton.ClientConfig) (interface{}, error) {
+					return identity.NewClient(config)
+				},
+			},
+			&testutils.StepAPICall{
+				StateBagKey: "identity",
+				CallFunc: func(client interface{}) (interface{}, error) {
+					c := client.(*identity.IdentityClient)
+					ctx := context.Background()
+
+					input := &identity.CreateRoleInput{
+						Name: testRoleName,
+					}
+					role, err := c.Roles().Create(ctx, input)
+					newRole = role
+
+					// Allow a few seconds for the new role
+					// to propagate within CloudAPI.
+					time.Sleep(5 * time.Second)
```

Comment:
I don't have a better idea, sadly. I did try with a retry, but while polling `Roles().Get()` works fine, then setting tags through `SetRoleTags()` would fail anyway. Unless we wait, a bit... This is driving me crazy a bit :(

Reply:
yeah such is eventual consistency!